Initial Blockout and Player Controller

HaydenHefford opened the pull request:
Both features are in the same pull request because I had to merge the Player Controller branch into my Blockout branch to actually make sure things looked alright.

Player controller:

A basic implementation that allows for 3D movement and camera control.

Blockout: A basic 3D blockout of Gabe's floorplan. This uses Godot's 3D gridmap system. This is largely devoid of detail right now, and what is there uses the gridmap system too. This can be subbed out for more detail on Sunday if we want, but thought it best to submit this now in case people need it for other testing, etc.

HaydenHefford committed:
Initial Blockout

Completed an initial blockout of each floor of the house using Godot's 3D gridmap system. No ceilings for each floor currently, as the stairs go too high to allow the floor above to double up as a ceiling. Will add a player controller and then return to this to find appropriate ceiling height once I've checked the scale.

HaydenHefford committed:
Basic player controls implemented

The player can move and look around in first person.

HaydenHefford committed:
Initial ceiling test

Added a ceiling to the ground floor of the blockout to test out how it feels with the player controller.

HaydenHefford committed:
Merge branch 'feature-player-movement' into feature-blockout

HaydenHefford committed:
Ceilings added to the blockout

Lee-RoyDobson commented on Scripts/Player_Scripts/player.gd:
Could speed and jump velocity be changed from constants to exported variables so that designers can change them in engine?

Lee-RoyDobson replied:
Can we change the input action from "ui_accept" to a jump action.

HaydenHefford committed:
Feedback Changes

Changed input action from "ui_accept" to "jump".

Made speed and jump velocity @export var rather than const.

Lee-RoyDobson reviewed (commented):
All good from me now those changes have been made.

bigmonmulgrew commented on Scripts/Player_Scripts/player.gd:
```diff
@@ -0,0 +1,37 @@
+func _ready():
+	Input.set_mouse_mode(Input.MOUSE_MODE_CAPTURED)
+
+func _physics_process(delta: float) -> void:
```
Physics process has three distinct functions, "ApplyGravity", "HandleJump" and "Handle Movement"

please abstract these into independent functions.

bigmonmulgrew commented on Scripts/Player_Scripts/player.gd:
```diff
@@ -0,0 +1,37 @@
+extends CharacterBody3D
+
+
+@export var SPEED = 5.0
```
all upper case is for constants.

These are exports, so not constant. They should be snake cases.

bigmonmulgrew reviewed (changes requested):
See other commetns.
Minor coding style and best practice change needed

Otherwise its tested and wortking.

HaydenHefford committed:
Abstracted functions and tweaked code style

Fixed the changes requested by Dave:

- abstracted the "jump", "apply_gravity", and "handle_movement" code into separate functions, all of which are then called in "_physics_process".

- changed variable names from all-caps to snake_case

bigmonmulgrew committed:
added underscores
